src/setContextLink.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  Operation,
+  NextLink,
+  FetchResult,
+} from './types';
+import * as Observable from 'zen-observable';
+
+import {
+  ApolloLink,
+} from './link';
+
+export default class SetContextLink extends ApolloLink {
+
+  constructor (private context?: any) {
+    super();
+  }
+
+  public request(operation: Operation, forward: NextLink): Observable<FetchResult> {
```

Comment:
@evanshauser I'm concerned about the current `forward` pattern since it could be optional? I'm a little confused at forward and how we even do things like enforcing passing a forward into a forward?